### ISSUE TITLE
Speech Command Recognition tutorial's usage of FC layer for M5 seems to contradict the original paper

### ISSUE THREAD

**Issue by kachayev** [pytorch/tutorials#1256]
The "Speech Command Recognition with torchaudio" tutorial uses fully connected linear layer as part of M5 model

```
    def __init__(self):
        ...
        self.fc1 = nn.Linear(2 * n_channel, n_output)
  
    def forward(self, x):
        ...
        x = self.fc1(x)
        return F.log_softmax(x, dim=2)
```

The tutorial states

> The specific architecture is modeled after the M5 network architecture described in this paper.

The original paper ["Very deep convolutional neural networks for raw waveforms"](https://arxiv.org/pdf/1610.00087.pdf) does not have a linear level in the described architecture (it does convolutions + pooling -> avg pooling -> softmax). The paper also mentions comparison of proposed architecture with the architecture that relies on FC for classification, quoting

>  However,  FC layers do not improve test accuracy ...

I would suggest either of the following solutions:
* remove FC layer
* rename model from M5 to `M5-fc` and mention the difference between M5-fc and M5
* show both models and do comparison as a part of the tutorial

Happy to PR changes. CC @vincentqb WDYT?

**Comment by vincentqb** [comment]
Thanks for the comment! Either of the first two options are fine to me. 
- If removing the FC layer makes the model simpler while maintaining similar convergence and notebook runtime, then that would be my preferred option.
- Otherwise, a simple fix would be to just add a note as a comment in the model definition about the extra fully connected layer, without even the need to rename the model.

**Comment by kachayev** [comment]
Thanks for the reply, @vincentqb! I'm going to compare 2 models and will post the result here so we can compare and decide the course of further actions.

**Comment by svekars** [comment]
Closing this as it looks like it was resolved. Feel free to reopen if needed.
